server/src/main/java/org/elasticsearch/common/settings/FallbackSecureSettings.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.settings;
+
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.common.hash.MessageDigests;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * An implementation of secure settings that fall back to the YML settings.
+ *
+ * WARNING: this is a temporary class to be used only in the context of Stateless. It applies only to YML settings with a predetermined
+ * prefix. Other secure settings do not fall back to YML settings.
+ */
+public class FallbackSecureSettings implements SecureSettings {
+    static final String FALLBACK_PREFIX = "insecure.";
+    static final Setting.AffixSetting<String> FALLBACK_SETTINGS = Setting.prefixKeySetting(
+        FALLBACK_PREFIX,
+        (key) -> Setting.simpleString(key, Setting.Property.NodeScope)
+    );
+
+    private final SecureSettings secureSettings;
+    private final Settings settings;
+    private final Set<String> names;
+
+    private FallbackSecureSettings(Settings settings) {
+        if (DiscoveryNode.isStateless(settings) == false) {
+            throw new IllegalArgumentException("FallbackSecureSettings are supported only in stateless");
+        }
+
+        SecureSettings secureSettings = Settings.builder().put(settings, true).getSecureSettings();
```

Comment:
Stateless doesn't use the keystore, so this seems unnecessary (these will always be empty).